src/Compilers/Core/MSBuildTask/InteractiveCompiler.cs

```diff
@@ -214,7 +215,20 @@ private DotnetHost DotnetHostInfo
 
         protected abstract string ToolNameWithoutExtension { get; }
 
-        protected sealed override string ToolName => DotnetHostInfo.ToolNameOpt;
+        protected sealed override string ToolName
+        {
+            get
+            {
+                if (CoreClrShim.IsRunningOnCoreClr)
+                {
+                    return $"{ToolNameWithoutExtension}.dll";
+                }
+                else
+                {
+                    return $"{ToolNameWithoutExtension}.exe";
+                }
```

Comment:
This logic is shared between InterativeCompiler and ManagedCompiler. Consider factoring into a common method.